src/classes/ACCT_AccountMerge_TEST.cls

```diff
@@ -71,6 +71,9 @@ private with sharing class ACCT_AccountMerge_TEST {
     *   Addresses are merged.
     */
     static testMethod void testMergeTwoHHAccounts() {
+        //skip the test if Advancement is installed
+        if(ADV_PackageInfo_SVC.useAdv()) return;
```

Comment:
@weizsforg Thanks for making this change. Been thinking about this code pattern all day. Seems to me that we're basically hardcoding all of our unit tests to very specifically check for the Advancement package. Though that meets the current need it doesn't extend beyond Advancement. So, rather than check for `if(ADV_PackageInfo_SVC.useAdv())` I think it might be better to create a new more generic class in NPSP for `UTIL_PackageExtension_SVC` that has a method for `skipUnitTests()`. For now that method can return the value of `ADV_PackageInfo_SVC.useAdv()`, but it would allow us to modify a single bit of code to expand this usage beyond the Advancement package.

In addition, it might be helpful if `skipUnitTests()` could accept an Enum parameter to know which types of unit tests to skip. Some possible examples are:
```
UTIL_PackageExtension_SVC.skipUnitTests(UTIL_PackageExtension_SVC.TestType.Household);
UTIL_PackageExtension_SVC.skipUnitTests(UTIL_PackageExtension_SVC.TestType.Rollups);
UTIL_PackageExtension_SVC.skipUnitTests(UTIL_PackageExtension_SVC.TestType.Allocations);
UTIL_PackageExtension_SVC.skipUnitTests(UTIL_PackageExtension_SVC.TestType.Addresses);
```

I'm happy to talk through this more.

Reply:
Hey @force2b, any chance we could continue this way and open a technical debt item in our backlog to address this at a later date? We have been working on this literally for moths (the PR with the other side of this was opened in December), and would like to get it in as soon as we can, provided it meets all technical requirements. :-)

Reply:
I agree @ceiroa.  for now let's leave it as is.  @force2b I think this is something the NPSP team could consider doing in the future once they better define it.  But I don't think we should be asking Adv to do this at this time.

Reply:
Completely understand @ceiroa. I'm coming in a bit late on this. My primary concern is extensibility as we add support for additional external packages (thinking of the Archival Bridge package). Still, I think we're fine with this architecture in the short term. I don't see it creating any packaging issues if refactoring is required at a later point.